src/credits.c

```diff
@@ -1,39 +1,180 @@
 };
 
-struct unkStruct_203B340
+// size: 0x8
+struct unkCreditsStruct_203B344
 {
-    // size: 0x84
-    u32 unk0;
+    /* 0x0 */ u32 state; // state variable
+    u32 unk4; // timer?
 };
 
-extern void sub_80015C0(u32, u32);
+// size: 0xC
+struct CreditsData
+{
+    /* 0x0 */ s16 type;
+    s16 unk2;
+    s16 unk4;
+    /* 0x8 */ u8 *text;
+};
+
+extern struct unkCreditsStruct_203B340 *gUnknown_203B340;
+extern struct unkCreditsStruct_203B344 *gUnknown_203B344;
 
+extern struct UnkTextStruct2 gUnknown_80E4A10;
+extern struct UnkTextStruct2 gUnknown_80E4A28;
+extern const u32 gUnknown_80E4A40[];
+extern struct CreditsData* gCreditsTable[27];
 extern u16 gUnknown_80E5990[];
-extern struct unkStruct_203B344 *gUnknown_203B344;
-extern struct unkStruct_203B340 *gUnknown_203B340;
+
+extern void xxx_call_draw_string(s32, u32, const u8 *, u32, u32);
+
+extern void sub_80015C0(u32, u32);
 extern s32 sub_8001658(u32, u32);
+extern void sub_80073B8(u32);
+extern void sub_80073E0(u32);
+extern void sub_8099690(u32);
+extern void sub_8099A5C(u32, u32, u32);
+extern void sub_8099AFC(u32, u32, u32);
+extern u8 sub_8099B94();
 
+// Unsure if param_2 is a s32/u32/ptr
+s32 DrawCredits(s32 creditsCategoryIndex, s32 param_2)
+{
+    s32 i;
+    u32 counter;
+    s32 temp2;
+    struct CreditsData *cred;
+    u8 *srcText;
+    u8 *destText1;
+    u8 *destText2;
+    u8 curChar;
+    u8 buffer[128];
+    
+    gUnknown_203B340 = (struct unkCreditsStruct_203B340 *)MemoryAlloc(sizeof(struct unkCreditsStruct_203B340), 8);
+    gUnknown_203B340->unk60 = 0;
+    gUnknown_203B340->unk64 = param_2;
+    gUnknown_203B340->creditsCategoryIndex = creditsCategoryIndex;
+
+    for (i = 0; i < 4; i++)
+        gUnknown_203B340->unk0[i] = gUnknown_80E4A10;
+    gUnknown_203B340->unk0[0] = gUnknown_80E4A28;
+
+    ResetUnusedInputStruct();
+    sub_800641C(&gUnknown_203B340->unk0[0], 1, 1);
+    sub_8099690(2);
+    sub_80073B8(0);
+
+    cred = creditsCategoryIndex[gCreditsTable]; // WTF
+    counter = 0;
+    while (cred->text != NULL) {
+        srcText = cred->text;
+        if (cred->unk4 >= 0)
+            counter = cred->unk4;
+        else
+            counter -= cred->unk4;
+        temp2 = cred->unk2;
+
+        switch (cred->type) {
+            case 1:
+                destText1 = buffer;
+                while ((curChar = *srcText) != 0) {
+                    *destText1 = curChar;
+                    srcText++;
+                    destText1++;
+                }
+                *destText1 = 0;
+                srcText = buffer;
+                break;
+            case 2:
+                destText2 = buffer;
+                while ((curChar = *srcText) != 0) {
+                    *destText2 = curChar;
+                    srcText++;
+                    destText2++;
+                }
+                *destText2 = 0;
+                srcText = buffer;
+                break;
+        }
+
+        xxx_call_draw_string(temp2, counter, srcText, 0, 0);
```

Comment:
`xxx_call_draw_string` is `(s32 x, s32 y, u8 *text, u32, u32)` so: 

`temp2` -> x (coord) so `unk2` is x coord of that credit

`counter` -> y (coord) so `unk4` is the y offset of that credit